parsing/parsetree.mli

```diff
@@ -412,53 +459,63 @@ and binding_op =
     pbop_loc : Location.t;
   }
 
-(* Value descriptions *)
+(** {2 Value descriptions} *)
 
 and value_description =
     {
      pval_name: string loc;
      pval_type: core_type;
      pval_prim: string list;
-     pval_attributes: attributes;  (* ... [@@id1] [@@id2] *)
+     pval_attributes: attributes;  (** [... [\@\@id1] [\@\@id2]] *)
      pval_loc: Location.t;
     }
-
-(*
-  val x: T                            (prim = [])
-  external x: T = "s1" ... "sn"       (prim = ["s1";..."sn"])
+(** Values of type {!value_description} represents:
+    - [val x: T],                     when {!pval_prim} is [[]]
+    - [external x: T = "s1" ... "sn"] when {!pval_prim} is [["s1";..."sn"]]
 *)
 
-(* Type declarations *)
+(** {2 Type declarations} *)
 
 and type_declaration =
     {
      ptype_name: string loc;
      ptype_params: (core_type * (variance * injectivity)) list;
-           (* ('a1,...'an) t; None represents  _*)
+           (** [('a1,...'an) t; None] represents  [_]*)
```

Comment:
This comment means:
```ocaml
    - [Some [a1;...;an]] represents the list of type parameters [('a1,...'an) t]
   - [None] represents [_ t]
```

Reply:
I must say I am still a little bit confused, as this comment is supposed to explain the type of the field `ptype_params` and I don't see the relation with `Some [a1;...;an]` or `None`:
- `(core_type * (variance * injectivity)) list` has no mention of `option`
- If `[a1;...;an]` is a list of `core_type`s, is there a constraint on them so that they can only be of the form `Ptyp_var`? (In which case why aren't they a located string instead of a `core_type`?)
- adding an explanation of what `variance` and `injectivity` represents would be useful I think.

Reply:
There used to be `string loc option` instead of `core_type` here, it was removed in b56dc4b3df8d022b54f40682a9d5d4168c690413 in 4.02 (this is part of Leo's work on extensible variants) and no one fixed the comment since. Just remove the `; None represents _` part. Thanks!